frontends/web/src/routes/account/summary/chart.tsx

```diff
@@ -475,134 +293,280 @@ class Chart extends Component<Props, State> {
+    setFormattedData(data.chartDataDaily);
   }
-}
 
+  if (source === 'hourly' && prevChartDataHourly?.length !== data.chartDataHourly.length) {
+    lineSeries.current?.setData(data.chartDataHourly);
+    chart.current?.timeScale().fitContent();
+    setFormattedData(data.chartDataHourly);
+  }
+
+  if (prevChartFiat !== data.chartFiat) {
+    reinitializeChart();
+  }
+
+  if (prevHideAmounts !== hideAmounts) {
+    chart.current?.applyOptions({
+      leftPriceScale: {
+        visible: hideAmounts ? false : !isMobile,
+      }
+    });
+  }
+
+  useEffect(() => {
+    if (!chartInitialized.current) {
+      initChart();
+    }
+    return () => {
+      removeChart();
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useEffect(() => {
+    if (!chartInitialized.current) {
+      initChart();
+    }
+  });
 
-const HOC = translate()(Chart);
+  useEffect(() => {
+    const { utcYear, utcMonth, utcDate, from, to } = getUTCRange();
+
+    switch (chartDisplay) {
+    case 'week': {
+      from.setUTCDate(utcDate - 7);
+      chart.current?.timeScale().setVisibleRange({
+        from: from.getTime() / 1000 as UTCTimestamp,
+        to: to.getTime() / 1000 as UTCTimestamp,
+      });
+      break;
+    }
+    case 'month': {
+      from.setUTCMonth(utcMonth - 1);
+      chart.current?.timeScale().setVisibleRange({
+        from: from.getTime() / 1000 as UTCTimestamp,
+        to: to.getTime() / 1000 as UTCTimestamp,
+      });
+      break;
+    }
+    case 'year': {
+      from.setUTCFullYear(utcYear - 1);
+      chart.current && chart.current.timeScale().setVisibleRange({
+        from: from.getTime() / 1000 as UTCTimestamp,
+        to: to.getTime() / 1000 as UTCTimestamp,
+      });
+      break;
+    }
+    case 'all': {
+      chart.current?.timeScale().fitContent();
+      break;
+    }
+    }
+  }, [source, chartDisplay]);
+
+  const {
+    lastTimestamp,
+    chartDataMissing,
+    chartFiat,
+    chartIsUpToDate,
+    chartTotal,
+    formattedChartTotal,
+  } = data;
+
+  const {
+    toolTipVisible,
+    toolTipValue,
+    toolTipTop,
+    toolTipLeft,
+    toolTipTime,
+  } = tooltipData;
+
+  const hasDifference = difference && Number.isFinite(difference);
+  const disableFilters = !hasData || chartDataMissing;
+  const disableWeeklyFilters = !hasHourlyData || chartDataMissing;
+  const showMobileTotalValue = toolTipVisible && !!toolTipValue && isMobile;
+  const chartFiltersProps = {
+    display: chartDisplay,
+    disableFilters,
+    disableWeeklyFilters,
+    onDisplayWeek: displayWeek,
+    onDisplayMonth: displayMonth,
+    onDisplayYear: displayYear,
+    onDisplayAll: displayAll,
+  };
 
-export { HOC as Chart };
+  const chartHeight = `${!isMobile ? height : mobileHeight}px`;
+
+  return (
+    <section className={styles.chart}>
+      <header>
+        <div className={styles.summary}>
+          <div className={styles.totalValue}>
+            {formattedChartTotal !== null ? (
```

Comment:
note: can be `undefined` needs to be fixed but wait for master and rebase

Reply:
did you rebase?

Reply:
Yes I did rebase, `formattedChartTotal` should now be either `null` or some value because #2850 now uses `success` and `error` response.

unrelated to this: I saw some other comments above, I thought I did address all change requests/review but will take another look and ping you if I think it is ready for review again, thanks :)